### PR TITLE
build(emitter): enable eslint

### LINKED ISSUE #4698: [DPG]: Enable ESLint
## Description

ESlint is not enabled for emitter. We should enable it as soon as possible.

- [x] Fix error in `npm run eslint` command
- [x] Add linter in build pipeline

### PR DESCRIPTION
**Pull request by archerzz**
# Description

- enable `npm run lint` which previously didn't work
- fix errors found by eslint
- add linter into build pipeline

resolve #4698 

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/TypeSpec.Extension/Emitter.Csharp/src/type/inputEnumTypeValue.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 export interface InputEnumTypeValue {
     Name: string;
+    /* eslint-disable @typescript-eslint/no-explicit-any -- TODO: can we specify the range of value types? */
     Value: any;
```
Comment (ArcturusZhang) [inline src/TypeSpec.Extension/Emitter.Csharp/src/type/inputEnumTypeValue.ts:7]: same here, `string | number | boolean`

File: src/TypeSpec.Extension/Emitter.Csharp/src/type/inputType.ts [hunk 2]
```diff
@@ -21,7 +21,9 @@ export interface InputLiteralType extends InputType {
     Kind: InputTypeKind.Literal;
     Name: InputTypeKind.Literal; // literal type does not really have a name right now, we just use its kind
     LiteralValueType: InputType;
+    /* eslint-disable @typescript-eslint/no-explicit-any -- TODO: can we specify the range of value types? */
     Value: any;
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply src/TypeSpec.Extension/Emitter.Csharp/src/type/inputType.ts:25]: or could we use unknown?

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts [hunk 3]
```diff
@@ -85,6 +84,7 @@ function isSchemaProperty(context: SdkContext, property: ModelProperty) {
     return !(headerInfo || queryInfo || pathInfo || statusCodeInfo);
 }
 
+/* eslint-disable @typescript-eslint/no-explicit-any -- A special recursion */
```
Comment (ArcturusZhang) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts:87]: if this rule has been disabled, we should remove this line, same to others

**Review by ArcturusZhang — APPROVED** [review]

File: src/TypeSpec.Extension/Emitter.Csharp/src/type/inputType.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (archerzz) [reply src/TypeSpec.Extension/Emitter.Csharp/src/type/inputType.ts]: I tried. That will cause some cascading changes which we'd better handle them later.

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts (hunk 3 above)
Comment (archerzz) [reply src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts]: removed

**Pull request merged**